nixos/modules/virtualisation/qubes.nix

```diff
@@ -0,0 +1,52 @@
+{lib, pkgs, config, ...}:
+let
+  inherit (lib) mkEnableOption mkIf;
+  cfg = config.virtualisation.qubes;
+  # FIXME: Xen build is broken with python 3.12+ due to https://github.com/NixOS/nixpkgs/issues/253751
+  pythonPackages = pkgs.python311Packages;
+in {
+  # Bug:
+  #   Loopback vchan connection not supported
+  #   FATAL: vchan initialization failed
+  # Solution: Xenstore has no set domid
+  #   It can be set manually using
+  #   `run0 xenstore-write "/domid" 0`
+  #   However it should always be set by default by something,
+  #   and I am not sure how/where.
+
+  # FIXME: Should it be qubes-dom0 or smth?
+  options.virtualisation.qubes = {
+    enable = mkEnableOption "Qubes";
+  };
+  config = mkIf cfg.enable {
+    users.groups.qubes = { };
+    users.users.qubes = {
+      isSystemUser = true;
+      group = "qubes";
+    };
+    systemd.tmpfiles.packages = [pythonPackages.qubes-core-admin];
+    systemd.tmpfiles.settings."10-qubes"."/var/log/qubes".d = {
+      user = "qubes";
+      group = "qubes";
+      mode = "700";
+    };
+    systemd.packages = [
+      # qubesdb
+      pkgs.qubes-core-qubesdb
+      # qubesd
+      pythonPackages.qubes-core-admin
+      # qrexec-agent, qrexec-fork-server
+      pythonPackages.qubes-core-qrexec.agent
+    ];
+
+    # Relevant for domU, dom0 uses qubes-db-dom0
+    systemd.services.qubes-db.enable = false;
+
+    environment.systemPackages = [
+      # FIXME: qrexec-policy-exec system path is hardcoded in qubes-core-qrexec.util
+      pythonPackages.qubes-core-qrexec
+      # FIXME: qubes-core-admin-client system path is hardcoded in qubes-core-admin
+      pythonPackages.qubes-core-admin-client
```

Comment:
Those packages (qubes-core-admin, qubes-core-admin-client) can't be merged, thus we need to deal with circular dependency somehow.
We may use PATH and add qubes-core-admin-client to the PATH of qubes-core-admin, but I don't think that will do.

Reply:
Cycles are resolved